### DISCUSSION TITLE
2009/05/17/%E5%88%9D%E8%AF%BB%E3%80%8A%E5%A4%A7%E8%AF%9D%E8%AE%BE%E8%AE%A1%E6%A8%A1%E5%BC%8F%E3%80%8B

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#261] [Announcements]
# 2009/05/17/%E5%88%9D%E8%AF%BB%E3%80%8A%E5%A4%A7%E8%AF%9D%E8%AE%BE%E8%AE%A1%E6%A8%A1%E5%BC%8F%E3%80%8B

昨天把新买的《大话设计模式》翻了一下，发现它在介绍面向对象编程时居然使用了和我同样的例子。我在《我和LabVIEW》中使用的一个例子是编写程序让一只小狗叫一声，再让一只小鸡叫一声；《大话设计模式》是让小狗和小猫叫一声，而它又注释说这个例子改变自《编程的奥秘》一书。看来大家的想法经常是会有类似的。这本书的作者也是先在博客上发文章，受到大家欢迎后才整理成书的。

https://blog.qizhen.xyz/2009/05/17/%E5%88%9D%E8%AF%BB%E3%80%8A%E5%A4%A7%E8%AF%9D%E8%AE%BE%E8%AE%A1%E6%A8%A1%E5%BC%8F%E3%80%8B

**Comment by ruanqizhen** [comment]
之前博客下的留言：

杰
2009年05月17日 23:52 编辑
看过Design Pattern的书，相当的晦涩。坚持看完了，不过回头想想，收获并不大，或许很多收获是潜移默化的吧，这玩意就是对干活时的思路、习惯有启发。

回复
永青
2009年05月17日 13:40 编辑
大哥，本来就没几种例子。。。

回复
